keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,1084 @@
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Disabled
+
+Please see the proposed timelines for this removal in the 
+[Design Details](#design-details) section. To move from the disabled to the 
+removed stage the following criteria needs to be met:
+
+- All code and tests to disable gitRepo volume driver by default is implemented
+- Two versions have passed since introducing the default disablement of gitRepo volume driver (to address version skew)
+- All feedback on usage/changed behavior, provided on GitHub issues has been addressed
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+- If you are upgrading a cluster with no pods that have gitRepo volume then no
+action is required
+- If you are upgrading a cluster with pods that have gitRepo volume then there
+are a few options:
+  - Enable the feature-gate GitRepoVolumeDriver
+  - Migrate the workloads to use [git-sync](https://github.com/kubernetes/git-sync)
+  - Migrate the workloads to use emptyDir + initContainer to sync the git repo
```

Comment:
xref to the kubectl commands later?

Reply:
I meant the `get pods ... | jq ...` command to learn if there are pods which need to be migrated off of `gitRepo`

Reply:
Ah sure will add that here.

Reply:
Done.